specification.markdown

```diff
@@ -749,6 +749,56 @@ type: Unsigned Integer
 
 description: The minimum `DocType` version an `EBML Reader` has to support to read this `EBML Document`. The value of the `DocTypeReadVersion Element` MUST be less than or equal to the value of the `DocTypeVersion Element`.
 
+### DocTypeExtension Element
+
+name: `DocTypeExtension`
+
+path: `0*(\EBML\DocTypeExtension)`
+
+id `0x4281`
+
+minOccurs: 0
+
+type: `Master Element`
+
+description: A `DocType` extension adds extra `Elements` to the main `DocType`+`DocTypeVersion` tuple it's attached to. An `EBML Reader` MAY understand these extra `Elements`. It MAY be used to iterate between experimental `Elements` before they are integrated in a regular `DocTypeVersion`. Reading one `DocTypeExtension` version of a `DocType`+`DocTypeVersion` tuple doesn't imply one should be able to read upper values of this `DocTypeExtension`.
+
+### ExtensionName Element
+
+name: `ExtensionName`
+
+path: `1*1(\EBML\DocTypeExtension\ExtensionName)`
+
+id `0x4283`
+
+minOccurs: 1
+
+maxOccurs: 1
+
+size: >0
+
+type: String
+
+description: The name of the `DocTypeExtension` to uniquely identify it from other `DocTypeExtension` of the same `DocType`+`DocTypeVersion` tuple.
```

Comment:
To me this sentence implies that a specific extension can only apply to a single `DocType`+`DocTypeVersion` tuple and that in order to apply to a different `DocTypeVersion` (of the same `DocType`) one would need a different `ExtensionName`. Similar wording for `ExtentionVersion` exists which confirms my believe.

I don't think this is the best way to handle extensions. I'd rather say that one `ExtensionName`+`ExtensionVersion` can apply to any `DocType`+`DocTypeVersion`, at least from the point of view of the EBML specs. I'd rather think it should be up to the `ExtensionName`+`ExtensionVersion` specs to specify if they can only be used with certain `DocType`+`DocTypeVersion` tuples.

This implies that `ExtensionName` must be unique across all `DocType`+`DocTypeVersion` tuples.

The reason I'm arguing for this is simple: it reduces complexity in an EBML Writer. For example, mkvmerge sets `DocTypeVersion` depending on which Matroska features are used to anywhere between 1 and 4 inclusively. If mkvmerge were to use a `DocTypeExtension`, there would have to be four different `ExtensionName`s specced, one for each of the four `matroska` `DocTypeVersion`s.